### DISCUSSION TITLE
v0.9

### DISCUSSION THREAD

**Discussion by Neko-Box-Coder** [Neko-Box-Coder/ssGUI#2] [General]
<hr /><em>This discussion was created from the release <a href='https://github.com/Neko-Box-Coder/ssGUI/releases/tag/v0.9'>v0.9</a>.</em>

**Comment by andreyminea** [comment]
Hi!
I would like to contribute to this project, are pull-request allowed ?

**Comment by Neko-Box-Coder** [comment]
Hi :wave: ,

I never really expected this but yeah I am more than happy to have contributors or even any suggestions on the project.
I am currently working on a new release that overhauls the entire documentation to be more user-friendly which will come in a week or so.

If you want to contribute anything, I think it is best to create an issue (about what you want to contribute) first and then we can have a discussion about it.

If it is anything very straightforward (so something like a typo or something), feel free to create a PR straightaway.

Thanks

**Reply by andreyminea** [reply]
Hello!
I didn't expected to get a response so fast. I've been trying to compile this code for 2 days, but it seems that it is not compatible with MSVC. Perhaps you can help me with the set-up and get up & running with it. You can find me on discord Thank you in advance.

**Reply by Neko-Box-Coder** [reply]
Hi,

Unfortunately, I don't have discord.
However, I do have some general steps that might help. If not, please create an issue (Something like "Support for MSVC") so that I can track it and also document it (since I don't use Visual Studio).

1. You need to have the latest (or recent enough) snapshot of SFML, from [here](https://artifacts.sfml-dev.org/by-branch/master/).
2. You need to be able to run the minimal example for SFML.
3. The graphics module must be accessible with `#include "SFML/Graphics.hpp"`
4. You want to download either the v0.92 release or clone from master.
5. Then you need to find a way to include the ssGUI files. You can either:
  a. Create a CMake file (or use the one you already have) to generate the visual studio project or
  b. Add the include directory to your project include directories. Then add all the sources of ssGUI.

Hope this helps, if not I can try to answer and debug the errors with you if you create a new issue for this.

Thanks

**Reply by andreyminea** [reply]
Hi!
So I managed yesterday to roughly follow these steps and build & run. I used the SFML 2.5 from their website. I resolved some of the issues regarding the compatibility with SFML 2.5 just to be able to run an example (I will switch to a latest build from the link indicated by you soon). I managed to run an example, but it crashes when trying to click on window, probably it will be fixed with the latest of SFML.

I do have a quick question regarding your development env. Are you using Linux, or VSCode with minGw compiler on windows?

I want to switch to Visual studio solution files as it will be faster for me to debug and  develop. Perhaps temporarily I could make a separate branch as it will be easier for others to fork and get up & running. I hope this will not be a problem.
As a second thing I see that there are a couple of old files that perhaps need an update from you, but I can't tell.

Hopefully I will be able to resolve all the issue and make some pull requests soon. I have some ideas regarding some changes and decoupling the SFML Backend.

Thanks, and let me know about the question about the env and if a separate branch with VS proj is ok with you.

**Reply by Neko-Box-Coder** [reply]
I am assuming the example you ran is for SFML, not including any of ssGUI stuff.

If so, I am surprised the snapshot from SFML is not working on Windows, after all this is the main OS they are supporting.
It is really important to use the same compiler and version and it could potentially be the cause of the crash.

I am using Linux with GCC (So the Debian GCC snapshot) with vscode, but ideally the compiler shouldn't matter. (As in my code should work fine all all compilers).

You can generate a visual studio project using CMake, you can take a look at the [CMakeLists.txt](https://github.com/Neko-Box-Coder/ssGUI/blob/main/src/CMakeLists.txt) I have that has plenty of comments which you can reference from. I also have sub-modules under the `src` directory which I totally forgot and you would need to manually include those files as well if you are not going down the cmake route.

As for a separate branch, I would imagine the only thing different from the master branch would be an extra .sln file right?
Unless there's a lot of generated files from Visual Studio, I don't quite see the point of doing it because it would need to be merged from master constantly.

But for now, I can create a separate branch for the next ssGUI release for MSVC when I do the new release if you really want to.

One thing I forgot to mention, if you want to run ssGUI, you will need to copy the src/resources folder to the location of the executable as it contains the default assets used by ssGUI. And it will crash (with debug messages) I think if it can't find the default resources.

Finally, I will create a issue and we can continue our conversation there.

Thanks

**Reply by andreyminea** [reply]
I was running one of your examples. I made to make your code & sfml compile

**Reply by andreyminea** [reply]
> As for a separate branch, I would imagine the only thing different from the master branch would be an extra .sln file right?

Yes, basically I will try to keep the structure like it is

**Reply by Neko-Box-Coder** [reply]
Oh okay, then it could be something wrong on my end.
Please continue the conversation [here](https://github.com/Neko-Box-Coder/ssGUI/issues/3) for easier tracking
